scripts/travis/exim-setup.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/sh -e
+#
+# ### This is a script to setup an exim smtp server for testing rlm_smtp
+#
+
+#
+# Declare the important path variables
+#
+
+# Base Directories
+BASEDIR=$(git rev-parse --show-toplevel)
+BUILDDIR="${BASEDIR}/build/ci/exim4"
+
+# Directories for exim processes
+RUNDIR="${BUILDDIR}/run"
+MAILDIR="${BUILDDIR}/mail"
+MAILDELIVERYDIR="${BUILDDIR}/mail_delivery_test"
+LOGDIR="${BUILDDIR}/eximlog"
+SPOOLDIR="${BUILDDIR}/spool"
+CERTDIR="${BUILDDIR}/certs"
+CERTSRCDIR="${BASEDIR}/raddb/certs"
+PASSWORD="whatever"
+
+# Important files for running dovecot
+CONF="${BUILDDIR}/exim.conf"
+
+# Used for supporting files
+TRAVISDIR="${BASEDIR}/scripts/travis"
+
+#
+# Prepare the directories and files needed for running exim
+#
+
+# Stop any currently running exim instance
+echo "Checking for a running exim instance"
+if [ -e "${RUNDIR}/exim.pid" ]
+then
+	echo "Stopping the current exim instance"
+	kill "$(cat ${RUNDIR}/exim.pid)"
```

Comment:
Maybe should be good force using SIGKILL. e.g: `kill -9 "$(cat ${RUNDIR}/exim.pid)"`

Reply:
No.  That can cause the sockets to remain bound for a period and prevent exim being restarted.  You only use `kill -9` if the process is completely unresponsive.